addons/web/models/ir_http.py

```diff
@@ -122,7 +122,7 @@ def session_info(self):
             # but is still included in some other calls (e.g. '/web/session/authenticate')
             # to avoid access errors and unnecessary information, it is only included for users
             # with access to the backend ('internal'-type users)
-            menus = self.env['ir.ui.menu'].load_menus(request.session.debug)
+            menus = self.env['ir.ui.menu'].with_context(request.session.context).load_menus(request.session.debug)
```

Comment:
Mind to only update the lang?
```suggestion
            menus = self.env['ir.ui.menu'].with_context(lang=request.session.context['lang']).load_menus(request.session.debug)
```